modules/programs/offlineimap.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ../lib/dag.nix { inherit lib; };
+with import ../lib/mail.nix { inherit lib config; };
+
+let
+  cfg = config.programs.offlineimap;
+
+  accounts =
+    filter (a: a.notmuch.enable)
+    (attrValues config.accounts.email.accounts);
+
+  # generates a script to fetch only a specific account
+  genOfflineImapScript = {name, ...} @ account:
+    pkgs.writeShellScriptBin "offlineimap-${name}" ''
+      ${pkgs.offlineimap}/bin/offlineimap -a${account.name} -c "$XDG_CONFIG_HOME/offlineimap/config" $@
+    '';
+
+  localTypeStr = account:
+    if isGmail account then "GmailMaildir" else "Maildir";
+
+  remoteTypeStr = account:
+    if isGmail account then "Gmail" else "IMAP";
+
+  remoteHostStr = account:
+    if account.imap.host != null then "remotehost = "+ account.imap.host else "";
+
+  loginCommandStr = account:
+    "remoteuser = ${account.userName}";
+
+  passwordCommandStr = account:
+  if account.passwordCommand != null then
+  ''remotepasseval = get_pass("${account.name}", "${toString account.passwordCommand}")''
+  else "";
+
+  postSyncHookStr = account:
+    if (builtins.stringLength account.offlineimap.postSyncHookCommand > 0) then
+        "postsynchook = ${account.offlineimap.postSyncHookCommand}"
```

Comment:
Hmm, if the command is multiple lines will this not confuse offlineimap? The config file would contain something like
```ini
postsynchook = cmd1
cmd2
cmd3
```
Perhaps better to write the command to a shell script (`writeScript`) and then put the path to this script here?

Reply:
I prefer to have the command in the config file rather than having another level of indirection.
The user can deal with it himself or we can revisit if there are problems ?